### DISCUSSION TITLE
line-height from custom pygments theme being ignored

### DISCUSSION THREAD

**Discussion by asmeurer** [pradyunsg/furo#438] [Potential Issue] [answered]
In our custom pygments theme for SymPy, we set the `line-height` attribute using `_pre_style`: https://github.com/sympy/sympy/blob/68c37df362e8585d72fc7ef490013bb8eff16e3e/doc/src/_pygments/styles.py#L25-L29

However, this is being ignored by the docs
<img width="428" alt="Screen Shot 2022-05-10 at 5 58 33 PM" src="https://user-images.githubusercontent.com/71486/167743194-00b7b8f0-70fb-4e98-81a3-ee0688a1ffc1.png">

We need this so that Unicode pretty printed output doesn't add extra space between lines. See for example https://docs.sympy.org/dev/tutorial/printing.html#unicode-pretty-printer and compare to our previous theme which set this attribute correctly https://docs.sympy.org/latest/tutorial/printing.html#unicode-pretty-printer

I can work around this by adding it to the custom.css, but it seems preferable to include it in the pygments theme since pygments allows this.

**Comment by pradyunsg** [comment]
Hmm... The reason this line exists in the Furo is because the value set by the default/most pygments themes results in extremely "tight" code blocks, when using the system font stack. It works fine if you use `monospace` or a code font with a lot of vertical space in its glyphs.

However, that is not the case for MacOS, Windows and Ubuntu when using the system font stack (at least, as I remember from the last time I checked). I just tried removed that but looking at the output from that reminded me why I had added this. :)

---

(current, with the line-height override)
<img width="753" alt="Screenshot 2022-05-11 at 20 53 47" src="https://user-images.githubusercontent.com/3275593/167934411-e60c33f7-ef29-4def-a0da-4a9385d3886c.png">

---

("respecting" pygments' line-height)
<img width="753" alt="Screenshot 2022-05-11 at 20 53 37" src="https://user-images.githubusercontent.com/3275593/167934425-960b0505-3298-40e3-920a-89e65c8c957b.png">

**Comment by asmeurer** [comment]
I can understand the tighter code is undesirable in most cases. We need it just so that our pretty printing looks good. Maybe we should consider whether that too negatively affects everything else. 

I actually forgot about font variability here. Our docs look best with DejaVu Sans Mono, because that's the best font for Unicode pretty printing, but it only uses that if you have it installed. 
This reminds me that for the SymPy docs we were planning to make a webfont for DejaVu Sans Mono so that it is guaranteed to be used by everyone on our docs. If we do that, that would remove the font variability from the equation. 

If you don't have DejaVu Sans Mono installed, here's screenshots showing why we need this. Personally I don't think the tightness is too bad for [everything else](https://output.circle-artifacts.com/output/job/7b8a9d9e-8535-4b41-8c48-89e3aec461ec/artifacts/0/doc/_build/html/tutorial/printing.html#unicode-pretty-printer), but I guess that's a personal judgement call. 
<img width="458" alt="Screen Shot 2022-05-11 at 2 11 26 PM" src="https://user-images.githubusercontent.com/71486/167939271-7500b122-19cf-464c-8f67-a1e6e9578c47.png">
<img width="454" alt="Screen Shot 2022-05-11 at 2 11 58 PM" src="https://user-images.githubusercontent.com/71486/167939288-7a967803-702e-48b7-a3b5-b5d625bff4c6.png">

Would be nice if we could get the best of both worlds somehow, but I don't know if that's possible.

**Comment by pradyunsg** [comment, marked as answer]
> Would be nice if we could get the best of both worlds somehow, but I don't know if that's possible.

I'm not sure either TBH.

https://github.com/pradyunsg/furo/blob/5abeb9fc962aee61eda02e0c75e872004635e9d5/src/furo/assets/styles/content/_code.sass#L40

This line still exists in Furo for now. I've filed https://github.com/pradyunsg/furo/issues/720 for further discussion on this, and for tracking a fix for this.

**Comment by kdeldycke** [comment]
I can confirm the issue.

I have a dozen of [unicode-based tables](https://kdeldycke.github.io/click-extra/table.html#table-formats) that are rendered into code blocks, and the wrong line height makes the lines of tables appears to be disjoint.

Here is a screenshot:

<img width="770" height="840" alt="Screenshot 2025-11-06 at 22 29 43" src="https://github.com/user-attachments/assets/e8c15385-aa66-43e6-8676-61458a536626" />

For context, this has been produced with the following dependencies:
```text
 + accessible-pygments==0.0.5
 + alabaster==1.0.0
 + babel==2.17.0
 + beautifulsoup4==4.14.2
 + docutils==0.21.2
 + furo==2025.9.25
 + imagesize==1.4.1
 + jinja2==3.1.6
 + markdown-it-py==3.0.0
 + markupsafe==3.0.3
 + mdit-py-plugins==0.5.0
 + mdurl==0.1.2
 + myst-parser==4.0.1
 + packaging==25.0
 + pygments==2.19.2
 + pyyaml==6.0.3
 + roman-numerals-py==3.1.0
 + snowballstemmer==3.0.1
 + soupsieve==2.8
 + sphinx==8.2.3
 + sphinx-autodoc-typehints==3.5.2
 + sphinx-basic-ng==1.0.0b2
 + sphinx-copybutton==0.5.2
 + sphinx-design==0.6.1
 + sphinx-issues==5.0.1
 + sphinxcontrib-applehelp==2.0.0
 + sphinxcontrib-devhelp==2.0.0
 + sphinxcontrib-htmlhelp==2.1.0
 + sphinxcontrib-jsmath==1.0.1
 + sphinxcontrib-mermaid==1.0.0
 + sphinxcontrib-qthelp==2.0.0
 + sphinxcontrib-serializinghtml==2.0.0
 + sphinxext-opengraph==0.13.0
 + typing-extensions==4.15.0
```
